Practicum/Pract. 04/zad1.cpp

```diff
@@ -0,0 +1,267 @@
+#include <iostream>
+#include <fstream>
+
+constexpr char FILE_NAME[] = "Offers.bin";
+constexpr char FILE_NAME_PERFECT[] = "PerfectOffers.bin";
+
+const unsigned NAME_LENGHT = 26;
+
+void clearCin()
+{
+	std::cin.clear();
+	while (std::cin.get() !='\n')
+	{
+
+	}
+}
+
+struct WorkOffer
+{
+	char companyName[NAME_LENGHT];
+	unsigned programersCount;
+	unsigned vacationDaysCount;
+	long long totalPay;
+
+};
+void ReadOfferFromConsole(WorkOffer& offer)
+{
+	clearCin();
+	char buffer[NAME_LENGHT];
+	do
+	{
+		
+			if (!std::cin)
+			{
+				clearCin();
+			}
+			std::cout << "Enter company name,with 25 symbols max: ";
+			std::cin.getline(buffer, NAME_LENGHT);
+		
+	} while (!std::cin);
+	strcpy_s(offer.companyName, buffer);
+
+
+	unsigned programmers = 0;
+	do
+	{
+
+		if (!std::cin || programmers > 100)
+		{
+			clearCin();
+		}
+		std::cout << "Enter programmersCount between 0 and 100 : ";
+		std::cin >> programmers;
+
+	} while (!std::cin || programmers > 100);
+	offer.programersCount = programmers;
+
+	unsigned vacationDays = 0;
+	do
+	{
+
+		if (!std::cin || vacationDays > 100)
+		{
+			clearCin();
+		}
+		std::cout << "Enter vacation days between 0 and 100 : ";
+		std::cin >> vacationDays;
+
+	} while (!std::cin || vacationDays > 100);
+	offer.vacationDaysCount = vacationDays;
+
+	unsigned totalPayVar = 0;
+	do
+	{
+
+		if (!std::cin || totalPayVar > 100)
+		{
+			clearCin();
+		}
+		std::cout << "Enter totalPay  between 0 and 100: ";
+		std::cin >> totalPayVar;
+
+	} while (!std::cin || totalPayVar > 100);
+	offer.totalPay = totalPayVar;
+
+}
+
+void printWorkOffer(const WorkOffer& offer)
+{
+	std::cout << "Company name: " << offer.companyName << "\n";
+	std::cout << "Programes count: " << offer.programersCount << "\n";
+	std::cout << "vacationDaysCount: " << offer.vacationDaysCount << "\n";
+	std::cout << "totalPay: " << offer.totalPay << "\n";
+
+}
+
+void filterOffers(const char* filePath, long long minSalary)
+{
+	WorkOffer readWorkOffer;
+	std::ifstream in(filePath,std::ios::binary | std::ios::in);
+
+	if (!in)
+	{
+		std::cerr << "Error with opening for reading file " << filePath << "\n";
+		return;
+	}
+
+	std::cout << "The ones who have salary above " << minSalary << "are\n";
+	while (in.read(reinterpret_cast<char*>(&readWorkOffer),sizeof(readWorkOffer)))
```

Comment:
you can read the whole file with only one read operation? Why you are not doing it?